### PR TITLE
Fix: Fix git flow, improve usage of Ruff, and fix small issues

### PR DESCRIPTION
**Pull request by xmnlab**
## Pull Request description

* Fix git flow after the creation of the project
* Improve the usage of Ruff
* Remove unnecessary text in the README.md (template)
* Add documentation_url, so the user can define what is the url for the documentation 

## How to test these changes

* ```...```

## Pull Request checklists

This PR is a:
- [ ] bug-fix
- [ ] new feature
- [ ] maintenance

About this PR:
- [ ] it includes tests.
- [ ] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [ ] pre-commit hooks were executed locally.
- [ ] this PR requires a project documentation update.

Author's checklist:
- [ ] I have reviewed the changes and it contains no misspelling.
- [ ] The code is well commented, especially in the parts that contain more complexity.
- [ ] New and old tests passed locally.

## Reviewer's checklist

Copy and paste this template for your review's note:

```
## Reviewer's Checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [ ] I managed to test the new changes locally
- [ ] I confirm that the issues mentioned were fixed/resolved .
```

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -33,6 +33,15 @@ repos:
...
         - python
 
+    - id: ruff-isort
+      name: ruff-isort
+      entry: ruff check --select I001 --fix
+      language: system
+      pass_filenames: true
+      files: "./"
+      types:
+        - python
```
Comment (Saransh-cpp) [inline .pre-commit-config.yaml:43]: Given that this PR enables `I001` in `pyproject.toml`, is adding this required? I think the `ruff` hook defined below should do the work. If we want to make the `isort` errors "fixable", we can add `fixable = ["I001"]` to `pyproject.toml` to do the work for us.

File: pyproject.toml [hunk 2]
```diff
@@ -83,6 +83,7 @@ select = [
   "YTT", # flake8-2020
   "PL",  # PL
   "RUF", # Ruff-specific rules
+  "I001", # isort
 ]
```
Comment (Saransh-cpp) [inline pyproject.toml:87]:
```suggestion
]
fixable = ["I001"]
```

File: src/scicookie/profile.py [hunk 3]
```diff
@@ -13,8 +14,8 @@ class Profile:
     """Profile class that handles profiles defined in the .yaml files."""
 
     profile_name: str = ""
-    config: dict = {}  # noqa: RUF012
-    profiles_available: list = []  # noqa: RUF012
+    config: Dict[str, Any] = {}  # noqa: RUF012
+    profiles_available: List[str] = []  # noqa: RUF012
```
Comment (Saransh-cpp) [inline src/scicookie/profile.py:18]: Given that we support only Python 3.8+, this should ideally be -
```suggestion
    config: dict[str, Any] = {}  # noqa: RUF012
    profiles_available: list[str] = []  # noqa: RUF012
```

File: src/scicookie/profile.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
 """Profile handles "profiles" defined in the .yaml files."""
 from pathlib import Path
+from typing import Any, Dict, List
```
Comment (Saransh-cpp) [inline src/scicookie/profile.py:3]:
```suggestion
from __future__ import annotations

from pathlib import Path
from typing import Any
```

File: src/scicookie/{{cookiecutter.project_slug}}/.pre-commit-config.yaml [hunk 5]
```diff
@@ -40,16 +48,25 @@ repos:
...
-{%- if cookiecutter.use_ruff == 'ruff' %}
+{%- if cookiecutter.use_ruff == 'yes' %}
+    - id: ruff-isort
+      name: ruff-isort
+      entry: ruff check --select I001 --fix
+      language: system
+      pass_filenames: true
+      files: "./"
+      types:
+        - python
```
Comment (Saransh-cpp) [inline src/scicookie/{{cookiecutter.project_slug}}/.pre-commit-config.yaml:59]: Same comment here!

File: tests/smoke/base.sh [hunk 6]
```diff
@@ -40,21 +40,21 @@ export PATH=$(echo $PATH| sed -E "s/[^:]+\/micromamba\/[^:]+//g")
...
   COMMAND_PREFIX="hatch run"
 elif command -v maturin &> /dev/null; then
-  pip install -e .
+  pip install .
 elif [ "$(pip list|grep -c scikit_build_core)" -ne "0" ]; then
-  pip install -e .
+  pip install .
 else
     # use setuptools
   pip install --editable .
```
Comment (Saransh-cpp) [inline tests/smoke/base.sh:60]: Should `setuptools` also not have an editable install?

**Review by Saransh-cpp — COMMENTED** [review]
Looks amazing! Thanks for all the work, @xmnlab! Some minor discussion points below =

File: .pre-commit-config.yaml (hunk 1 above)
Comment (xmnlab) [reply .pre-commit-config.yaml:43]: oh I didn't know, thanks for point me to that. I will apply that, thank you so much.

Comment (xmnlab) [reply .pre-commit-config.yaml:43]: oh just saw that you already suggested that change .. just accepted it and push your changes. thanks

File: src/scicookie/profile.py (hunk 3 above)
Comment (xmnlab) [reply src/scicookie/profile.py:18]: I think that it just works for py3.9+

```
Python 3.8.1 | packaged by conda-forge | (default, Jan 29 2020, 14:55:04) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a: list[str] = []
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'type' object is not subscriptable
```

https://docs.python.org/3/library/typing.html#typing.Dict

Comment (xmnlab) [reply src/scicookie/profile.py:18]: oh got your point, it works with `from __future__ import annotations` (that comment was below XD)
thank you so much, I didn't know that. I will push your suggestions! thanks!

File: tests/smoke/base.sh (hunk 6 above)
Comment (xmnlab) [reply tests/smoke/base.sh:60]: you're right! good catch ... I just search by `-e` in the code .. I forgot to search bybout `--editable` XD

**Comment by xmnlab** [comment]
thank you so much @Saransh-cpp ! today I learned something new :)
appreciate your review

**Pull request merged**